community/modules/compute/schedmd-slurm-gcp-v6-nodeset-tpu/variables.tf

```diff
@@ -89,6 +89,22 @@ variable "zone" {
   type        = string
 }
 
+variable "zones" {
+  description = <<-EOD
+    Additional nodes in which to allow creation of partition nodes. Google Cloud
+    will find zone based on availability, quota and reservations.
+    EOD
+  type        = set(string)
+  default     = []
+
+  validation {
+    condition = alltrue([
+      for x in var.zones : length(regexall("^[a-z]+-[a-z]+[0-9]-[a-z]$", x)) > 0
```

Comment:
incorrect regex. There are zones with multiple digits, e.g. `europe-west10-a`.
Let's not make any assumptions, if there is any problems with zone, they will surface during plan (I hope).

Reply:
Good catch.  Fixed here and in the original code I copied from.

Reply:
IMHO, We better to not have any assumptions about what is correct zone name. The most I would recommend is just "consist of letters, digits, _ and -"

Reply:
I don't think underscores are allowed in zone names, so I left that out.